File: LeetCode/Evaluate Reverse Polish Notation.cpp
class Solution {
public:
	//处理逆波兰表达式
    int evalRPN(vector<string> &tokens) {
		if(tokens.size()==1) return stoi(*tokens.begin());
		stack<int> st;
		vector<string>::iterator it=tokens.begin();
		int x=0,y=0;
		while(it!=tokens.end()){
			if(*it=="+"||*it=="-"||*it=="*"||*it=="/"){
				y=st.top();st.pop();
				x=st.top();st.pop();
				if(*it=="+") st.push(x+y);
				else if(*it=="-") st.push(x-y);
				else if(*it=="*") st.push(x*y);
				else if(*it=="/") st.push(x/y);
			}
			else st.push(stoi(*it));
			it++;
		}
		return st.top();        
    }
};

File: LeetCode/EvaluateReversePolishNotation.java
package com.leetcode.algorithm;

import java.io.IOException;
import java.util.*;


public class EvaluateReversePolishNotation {
	public static int evalRPN1(String[] tokens) {
		int returnValue = 0;
		String operators = "+-*/";
 
		Stack<String> stack = new Stack<String>();
 
		for (String t : tokens) {
			if (!operators.contains(t)) {
				stack.push(t);
			} else {
				int a = Integer.valueOf(stack.pop());
				int b = Integer.valueOf(stack.pop());
				switch (t) {
				case "+":
					stack.push(String.valueOf(a + b));
					break;
				case "-":
					stack.push(String.valueOf(b - a));
					break;
				case "*":
					stack.push(String.valueOf(a * b));
					break;
				case "/":
					stack.push(String.valueOf(b / a));
					break;
				}
			}
		}
 
		returnValue = Integer.valueOf(stack.pop());
 
		return returnValue;
	}
	
    public static int evalRPN(String[] tokens){
    	int ans = 0;
    	Stack<String> stack = new Stack<String>();
    	String operators = "+-*/";
    	for(String tmp:tokens){
    		if(operators.contains(tmp)){
    			int a = Integer.valueOf(stack.pop());
    			int b = Integer.valueOf(stack.pop());
    			switch (tmp) {
				case "+":
					stack.push(String.valueOf(a+b));
					break;
				case "-":
					stack.push(String.valueOf(b-a));
					break;
				case "*":
					stack.push(String.valueOf(a*b));
					break;
				case "/":
					stack.push(String.valueOf(b/a));
					break;
				}
    		}
    		else{
    			stack.push(tmp);
    		}
    	}
    	ans = Integer.valueOf(stack.pop());
    	return ans;
    }
    
	public static void main(String[] args){
		String[] tokens = new String[] { "0", "3", "/" };
		System.out.println(evalRPN(tokens));
	}
}
